### PR TITLE
core: plat-ls: get HW unique key using OP-TEE CAAM driver

### PR DESCRIPTION
**Pull request by sahilnxp**
Previously HW Unique key on LS platforms came through ATF
via SMC, since we have CAAM driver available in OP-TEE
itself, will use that directly from now on.

Signed-off-by: Sahil Malhotra <sahil.malhotra@nxp.com>

### CODE REVIEW

File: core/arch/arm/plat-ls/main.c [hunk 1]
```diff
@@ -206,43 +206,3 @@ void main_secondary_init_gic(void)
 {
 	gic_cpu_init(&gic_data);
 }
-
-#ifdef CFG_HW_UNQ_KEY_REQUEST
-
-#include <types_ext.h>
-int get_hw_unique_key(uint64_t smc_func_id, uint64_t in_key, uint64_t size);
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-ls/main.c:213]: Should `core/arch/arm/plat-ls/ls_hw_unq_key_a64.S` be removed too?

Comment (sahilnxp) [reply core/arch/arm/plat-ls/main.c:213]: Yes, It should be removed. 
Updated the commit.
Thanks @jenswi-linaro

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by sahilnxp** [comment]
@jenswi-linaro  Tag updated.

**Pull request merged**